src/bl/commentFacade.ts

```diff
@@ -0,0 +1,117 @@
+            }
+        } catch (e) {
+            throw new NotFoundException(e)
+        }
+    }*/
+
+    static async getById(commentId: number): Promise<Comment> {
+        try {
+            const comment = await getRepository(Comment)
+                                 .createQueryBuilder('comment')
+                                 .leftJoinAndSelect('comment.card', 'card')
+                                 .leftJoinAndSelect('comment.user', 'user')
+                                 .where('comment.id = :commentId', { commentId })
+                                 .getOne()
+            if (comment) {
+                return comment
+            } else {
+                throw new NotFoundException('No Comment was found')
+            }
+        } catch (e) {
+            throw new NotFoundException(e)
+        }
+    }
+
+    static async delete(commentId: number): Promise<boolean> {
+        try {
+            const comment = await CommentFacade.getById(commentId)
+            const cardId = comment.card.id
+            delete comment.card
+            
+            await getRepository(Comment).removeById(commentId)
+
+            RealTimeFacade.sendEvent(commentDeleted(comment, cardId))
+
+            return true
+
+        } catch (e) {
+            throw new NotFoundException(e)
+        }
+    }
+
+    /*
+    static async update(commentReceived: Comment): Promise<void> {
+        try {
+            const commentToSave = ParamsExtractor.extract<Comment>(['content', 'createdDate'], 
+                                                                    commentReceived)
+            const repository = getManager().getRepository(Comment)
+            return repository.updateById(commentReceived.id, commentToSave)
+        } catch (e) {
+            throw new NotFoundException(e)
+        }
+    }*/
+
+    static async create(params: {}, cardId: number): Promise<Comment> {
+        try {
+            
+            let commentToCreate = new Comment()
+            let userId
+
+            // extract params
+            const properties = ['content', 'createdDate', 'userId']
+            properties.forEach(name => {
+                if (name === 'userId') {
+                    userId = params[name]
+                } else if (params.hasOwnProperty(name)) {
+                    commentToCreate[name] = params[name]
+                }
+            })
+            // we don't want the user to be able to send his own date
+            commentToCreate.createdDate = new Date()
+
+            // add into user
+            if (userId !== undefined) {
+                commentToCreate.user = await UserFacade.getById(userId)
+                UserFacade.addComment(commentToCreate, userId)
+            } else {
+                throw new MissingParameterException('userId (author) missing')
+            }
+            // add into card
+            commentToCreate.card = await CardFacade.getById(cardId)
```

Comment:
I guess comment.card is a reference to the Card, but it seems weird here to read. What does it do ?

Reply:
comment.card est de type Card, c'est les liens crées par typeorm pour acceder aux entitées